data/rulesets.ts

```diff
@@ -927,6 +928,12 @@ export const BattleFormats: {[k: string]: FormatsData} = {
 					types = originalForme.types;
 					if (baseSpecies.otherFormes) {
 						for (const formeid of baseSpecies.otherFormes) {
+							let skipForme = false;
+							if (baseSpecies.prevo) {
+								const prevo = dex.getSpecies(baseSpecies.prevo);
+								skipForme = prevo.evos.map(toID).includes(toID(formeid));
+							}
+							if (skipForme) continue;
 							const forme = dex.getSpecies(formeid);
 							if (forme.changesFrom === originalForme.name) {
 								types = types.concat(forme.types);
```

Comment:
```suggestion
						for (const formeName of baseSpecies.otherFormes) {
							if (baseSpecies.prevo) {
								const prevo = dex.getSpecies(baseSpecies.prevo);
								if (prevo.evos.includes(formeName)) continue;
							}
							const forme = dex.getSpecies(formeName);
							if (forme.changesFrom === originalForme.name) {
								types = types.concat(forme.types);
```

The `toID` mapping seems excessive...

Reply:
I disagree. It's there to ensure that typos (such as a miscapitalized O) in the data file don't get in the way of validation.

Reply:
In any case, I did this (locally) and pushed because the type of `Species#evos` was incorrect.

Reply:
> It's there to ensure that typos (such as a miscapitalized O) in the data file don't get in the way of validation.

This is why I added data tests, so we wouldn't have to worry about stuff like this...

Reply:
Oh yeah